src/app/utils/brc20.ts

```diff
@@ -0,0 +1,57 @@
+import { UTXO, brc20TransferEstimateFees, FungibleToken } from '@secretkeylabs/xverse-core';
+import BigNumber from 'bignumber.js';
+
+export type TransferEstimateResult = Awaited<ReturnType<typeof brc20TransferEstimateFees>>;
+
+export type Brc20TransferEstimateFeesParams = {
+  addressUtxos: UTXO[];
+  tick: string;
+  amount: number;
+  feeRate: number;
+  revealAddress: string;
+};
+
+export type SendBrc20TransferState = {
+  fungibleToken: FungibleToken;
+};
+
+export type ConfirmBrc20TransferState = {
+  recipientAddress: string;
+  estimateFeesParams: Brc20TransferEstimateFeesParams;
+  estimatedFees: TransferEstimateResult;
+  token: FungibleToken;
+};
+
+export type ExecuteBrc20TransferState = {
+  recipientAddress: string;
+  estimateFeesParams: Brc20TransferEstimateFeesParams;
+  token: FungibleToken;
+};
+
+export type Brc20TransferParams = {
+  recipient: string;
+  addressUtxos: UTXO[];
+  ticker: string;
+  amount: number;
+  transferFees: TransferEstimateResult;
+};
+
+export type Brc20TransactionParams = Omit<Brc20TransferParams, 'transferFees'> & {
+  seed: string;
+  feeRate: number;
+};
+
+export const getFeeValuesForBrc20OneStepTransfer = ({
+  commitChainFee,
+  revealChainFee,
+  transferChainFee,
+  revealServiceFee,
+  transferUtxoValue,
+}: TransferEstimateResult['valueBreakdown']) => {
+  const txFee = new BigNumber(commitChainFee).plus(revealChainFee).plus(transferChainFee);
+  const inscriptionFee = new BigNumber(revealServiceFee);
+  const totalFee = new BigNumber(txFee).plus(inscriptionFee);
+  const btcFee = new BigNumber(transferUtxoValue);
```

Comment:
This isn't really the btc fee. It's the sats value of the final UTXO that will hold the inscription.

Reply:
i'll rename this in another PR, when design is finalised for: https://linear.app/xverseapp/issue/ENG-2785/notice-for-an-non-taproot-address-when-sending-brc-20-tokens

Reply:
👍